core/controllers/reader.py

```diff
@@ -1576,12 +1939,25 @@ def get(self, exploration_id):
         self.values.update({
             'summaries': (
                 summary_services.get_displayable_exp_summary_dicts_matching_ids(
-                    recommended_exp_ids)),
+                    list(recommended_exp_ids))),
```

Comment:
Can we allow `Set` as an input to the called function?

Reply:
I don't think so because if we make the argument type as a set then we have to change the argument type of all the functions that are called inside the get_displayable_exp_summary_dicts_matching_ids method. Because then `recommended_exp_ids` will propagate as set over function to function not as list and now most of the functions are accepting list. Please let me know if that clears the thing or i need to explain with references?

Reply:
Makes sense.